src/app/store/slices/plan/index.ts

```diff
@@ -169,18 +164,31 @@ export const planSlice = createSlice({
 
     builder.addCase(fetchSubscriptionThunk.fulfilled, (state, action) => {
       if (action.payload !== null) {
-        state.subscription = action.payload;
+        if (action.meta.arg.userType == UserType.Individual) {
+          state.individualSubscription = action.payload;
+          if (action.payload.type === 'subscription') {
+            state.individualPlan = action.payload.plan || null;
+          }
+        }
+        if (action.meta.arg.userType == UserType.Business) {
+          state.businessSubscription = action.payload;
+          if (action.payload.type === 'subscription') {
+            state.businessPlan = action.payload.plan || null;
+          }
+        }
```

Comment:
Same here, add the arg to a const with the corresponding Type to be able to read and understand this code in a more direct way without having to look for what type the payload is.